### PR TITLE
Document indirect data set usage in tests

### PR DESCRIPTION
**Pull request by Anthchirp**
and simplify code with `pytest.approx` and `Path` objects.

Just a note that I observed
```
tests/algorithms/integration/ssx/test_ellipsoid_integrate.py: 68 warnings
  /scratch/wra62962/files/dials/modules/dials/algorithms/profile_model/ellipsoid/indexer.py:50: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    h = np.array([int(floor(j + 0.5)) for j in hf[:, 0]], dtype=np.int).reshape(
```
which I assume is known.

Other things I noticed: calling the `run()` function from within the test does set up logging multiple times, so is not ideal. And neither is the use of `run_in_tmpdir`, which may cause race conditions as the working directory is essentially a shared global variable.

### CODE REVIEW

**Comment by jbeilstenedmands** [comment]
Hope you don't mind me adding some more stuff here, since the tests were failing anyway thought it would be good to try and fix all issues discussed.

File: algorithms/indexing/ssx/analysis.py [hunk 1]
```diff
@@ -8,10 +9,13 @@
 
 from scitbx.array_family import flex
 from xfel.clustering.cluster import Cluster
+from xfel.clustering.cluster_groups import unit_cell_info
 
 from dials.algorithms.clustering import plots as cluster_plotter
 from dials.util import tabulate
 
+logger = logging.getLogger("dials")
```
Comment (Anthchirp) [inline algorithms/indexing/ssx/analysis.py:17]:
```suggestion
logger = logging.getLogger("dials.algorithms.indexing.ssx.analysis")
```

File: algorithms/indexing/ssx/analysis.py [hunk 2]
```diff
@@ -102,6 +106,31 @@ def make_cluster_plots(large_clusters: List[Cluster]) -> dict:
...
+    threshold = math.floor((min_cluster_pc / 100) * len(crystal_symmetries))
+    large_clusters = [c for c in clusters if len(c.members) > threshold]
+    large_clusters.sort(key=lambda x: len(x.members), reverse=True)
+
+    if large_clusters:
+        logger.info(
+            f"""
+Unit cell clustering analysis, clusters with >{min_cluster_pc}% of the number of crystals indexed
+"""
+            + unit_cell_info(large_clusters)
```
Comment (Anthchirp) [inline algorithms/indexing/ssx/analysis.py:123]: could be integrated in the f-string?

File: tests/command_line/test_ssx_integrate.py [hunk 3]
```diff
@@ -1,81 +1,90 @@
...
-    run_integrate(
+    dials_data("cunir_serial", pathlib=True)
+    result = procrunner.run(
         [
-            indexed_refl,
-            indexed_expts,
-            "algorithm=stills",
+            "dev.dials.ssx_integrate",
+            str(ssx / "indexed.refl"),
+            str(ssx / "indexed.expt"),
```
Comment (Anthchirp) [inline tests/command_line/test_ssx_integrate.py:24]:
```suggestion
            ssx / "indexed.refl",
            ssx / "indexed.expt",
```
procrunner can deal with path objects

File: tests/command_line/test_ssx_integrate.py [hunk 4]
```diff
@@ -1,81 +1,90 @@
...
+        ],
+        working_directory=tmp_path,
     )
+    assert not result.returncode and not result.stderr
+    assert (tmp_path / "integrated_1.refl").is_file()
+    assert (tmp_path / "integrated_1.expt").is_file()
+    assert (tmp_path / "integrated_2.refl").is_file()
+    assert (tmp_path / "integrated_2.expt").is_file()
+    assert (tmp_path / "dials.ssx_integrate.html").is_file()
+    assert (tmp_path / "data.json").is_file()
```
Comment (Anthchirp) [inline tests/command_line/test_ssx_integrate.py:38]:
```suggestion
    assert tmp_path.joinpath("integrated_1.refl").is_file()
    assert tmp_path.joinpath("integrated_1.expt").is_file()
    assert tmp_path.joinpath("integrated_2.refl").is_file()
    assert tmp_path.joinpath("integrated_2.expt").is_file()
    assert tmp_path.joinpath("dials.ssx_integrate.html").is_file()
    assert tmp_path.joinpath("data.json").is_file()
```
I find `x.y().z()` easier to read than `(x / y).z()`

**Pull request merged**